### ISSUE TITLE
Nothing happens when choosing a dataset to load on Mac

### ISSUE THREAD

**Issue by vlad43210** [cmu-phil/tetrad#1710]
Hi,

I'm running Tetrad 7.6.0-0 on Mac OS 14 Sonoma. I tried making a data box > File > Load Data, chose my file, but nothing happens when I click "Choose." Any thoughts? I saw there are some security issues about Tetrad not seeing any files, but it is seeing the files, just nothing happens when I select one. I did go into security settings and grant java access to my Documents and Downloads folder.

**Comment by jdramsey** [comment]
Ah huh! I know the answer to this by now! OK, two things.

1. Could you switch to Tetrad 7.6.1?

2. Could you tell me what version of Java you are using? If I'm right, you're using 1.8, and you need to switch to one of the ore recent versions. You can tell by opening up a terminal and typing:

java -version

**Comment by vlad43210** [comment]
Thanks! I switched to Tetrad 7.6.1, uninstalled Oracle Java, and got OpenJDK 21.0.1. Now I see the Data Loader screen!

**Comment by jdramsey** [comment]
Yay!!! :-)

**Comment by jdramsey** [comment]
@magatz Sorry, I didn't get back to you. I was thinking. I think maybe what we'll need to do is stop supporting JDK 1.8 altogether and rewrite that bit of code that's problematic. It's only one method, and it's dragging everyone down. 1.8 is really a dinosaur JDK at this point, though some people have trouble switching away from it because they don't manage their own machines. (For instance, the may on servers they can't change the software on.) But I just watched a presentation by Amazon literally begging people to move to 17 and showing chart after chart of why.

So let me try to talk to some users who are still on 1.8 to see if there's any chance they can upgrade.

By the way if you happen to be one of those people (ahem!) please speak up.

**Comment by jdramsey** [comment]
Let me rephrase my previous comment (which I just deleted). The issue comes down to one method in the code, which could be rewritten, though rewriting it would make it incompatible with JDK 1.8. Some people are having difficulty installing a more recent version of the JDK, so we need to check to see if we can do that.

One thing you could try is to install Amazon Corretto's (Open)JDK 17, which is working for others.

**Comment by magatz** [comment]
thanks! I will give a try since I manage the RedHat server. Will keep you updated here

**Comment by jdramsey** [comment]
By the way we're going to release version 7.6.2 in a few days under (Corretto) JDK 17; a lot of the outstanding issues will be fixed.

**Comment by magatz** [comment]
Thanks! I will indeed check!
